impl/src/planning/subplanners.rs

```diff
@@ -279,6 +267,8 @@ impl<'planner> WorkspaceSubplanner<'planner> {
       }
     }
 
+    let mut root_ctx = None;
```

Comment:
This bit, and its devil children I dont fully like, it would be nice maybe to just grab the root contect without having to discover it through resolving deps.

I could not think of or find a way to do it, its probably ok for now, and maybe a better future approach might be to move to guppy as a dep